docs/content/docs/deployment/security/running-cluster.md

```diff
@@ -0,0 +1,292 @@
+cluster. For YARN setups, the cluster deployment phase can automatically distribute the keystore and 
+truststore files. For externally facing REST endpoints, the common name or subject alternative names in the
+certificate should match the node's hostname and IP address.
+
+For the externally facing REST endpoint, the common name or subject alternative names in the certificate
+should match the node's hostname and IP address.
+
+For more information, see the [documentation on YARN security](https://github.com/apache/hadoop/blob/trunk/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/YarnApplicationSecurity.md).
+
+## Setting Up SSL 
+
+The easiest way to set up SSL is to generate a dedicated public/private key pair and self-signed 
+certificate for the Flink deployment. The keystore and truststore are identical and contains
+only that key pair and certificate. 
+
+In an environment where operators are constrained to use firm-wide Internal Certificate Authorities 
+(cannot generate self-signed certificates), we recommend to still have a dedicated key pair and certificate
+for the Flink deployment, signed by that CA. However, the truststore must then also contain the
+CA's public certificate to accept the deployment's certificate during the SSL handshake (a requirement
+in the JDK truststore implementation).
+
+{{< hint danger >}}
+It is critical that you specify the fingerprint of the deployment certificate (via the `security.ssl.internal.cert.fingerprint` 
+setting), when it is not self-signed. This is to pin that certificate as the only trusted certificate 
+and prevent the truststore from trusting all certificates signed by that CA.
+{{< /hint >}}
+
+{{< hint info >}}
+Since internal connections are mutually authenticated with shared certificates, Flink can skip 
+hostname verification. This makes container-based setups easier.
+{{< /hint >}}
+
+### Tips for a YARN Deployment
+
+To secure the external REST endpoint, you need to issue the REST endpoint's certificate such that it 
+is valid for all hosts that the JobManager may get deployed to. This can be done with a wildcard DNS 
+name or by adding multiple DNS names.
+
+The easiest way to deploy keystores and truststore is via the YARN client's *ship files* option (`-yt`).
+Copy the keystore and truststore files into a local directory (i.e. `deploy-keys/`) and start the YARN 
+session like this: 
+
+```bash
+flink run -m yarn-cluster -yt deploy-keys/ flinkapp.jar
+```
+
+When deployed using YARN, Flink's web UI is accessible through the YARN proxy's "Tracking URL". To 
+ensure that the YARN proxy is able to access Flink's HTTPS URL, you need to configure the YARN proxy
+to accept Flink's SSL certificates. To do this, add the custom CA certificate into Java's default 
+truststore on the YARN proxy node.
+
+### Example SSL Setup for Standalone Deployment and Kubernetes Deployment
```

Comment:
Is this really specific for standalone or kubernetes? If not, I'd say it rather belongs to the SSL page instead or we just link to an external guide for the non-Flink specific parts.